common/djangoapps/terrain/ui_helpers.py

```diff
@@ -537,6 +537,15 @@ def click_link(partial_text, index=0):
 
 
 @world.absorb
+def click_button(data_attr, index=0):
+    xpath = '//button[text()="{button_text}"]'.format(
+        button_text=data_attr
+    )
+    world.browser.find_by_xpath(xpath)[index].click()
+    wait_for_js_to_load()
```

Comment:
Why did you put in wait_for_js_to_load? We try not to put in blanket waits like that, as they typically do not address the actual issue, and it is better to wait for a specific element expected on the page.

That being said, if this was necessary to get tests to pass, I don't care too much since they are Lettuce tests.